example_test.go

```diff
@@ -78,3 +79,147 @@ func Example() {
+	err = sender.Send(ctx, amqp.NewMessage([]byte("Hello!")))
+
+	var connErr *amqp.ConnError
+	if !errors.As(err, &connErr) {
+		log.Fatalf("unexpected error type %T", err)
+	}
+
+	// similarly, methods on session will fail in the same way
+	_, err = session.NewReceiver(ctx, "/queue-name", nil)
+	if !errors.As(err, &connErr) {
+		log.Fatalf("unexpected error type %T", err)
+	}
+
+	// methods on the connection will also fail
+	_, err = conn.NewSession(ctx, nil)
+	if !errors.As(err, &connErr) {
+		log.Fatalf("unexpected error type %T", err)
+	}
+}
+
+func ExampleSessionError() {
+	// *SessionErrors are returned when a session has been clossed.
+	// this error is propagated to all child Sender and Receiver instances.
+
+	// create connection
+	conn, err := amqp.Dial("amqps://my-namespace.servicebus.windows.net", &amqp.ConnOptions{
+		SASLType: amqp.SASLTypePlain("access-key-name", "access-key"),
+	})
+	if err != nil {
+		log.Fatal("Dialing AMQP server:", err)
+	}
+	defer conn.Close()
+
+	ctx := context.TODO()
+
+	// open a session
+	session, err := conn.NewSession(ctx, nil)
+	if err != nil {
+		log.Fatal("Creating AMQP session:", err)
+	}
+
+	// create a sender
+	sender, err := session.NewSender(ctx, "/queue-name", nil)
+	if err != nil {
+		log.Fatal("Creating sender link:", err)
+	}
+
+	// close the session before sending the message
+	session.Close(ctx)
+
+	// attempt to send message on a closed session
+	err = sender.Send(ctx, amqp.NewMessage([]byte("Hello!")))
+
+	var connErr *amqp.SessionError
+	if !errors.As(err, &connErr) {
+		log.Fatalf("unexpected error type %T", err)
+	}
+
+	// similarly, methods on session will fail in the same way
+	_, err = session.NewReceiver(ctx, "/queue-name", nil)
+	if !errors.As(err, &connErr) {
+		log.Fatalf("unexpected error type %T", err)
+	}
+}
+
+func ExampleDetachError() {
+	// *DetachErrors are returned by methods on Senders/Receivers after Close() has been called.
+	// it can also be returned if the peer has detached from the link. in this case, the *RemoteErr
+	// field should contain additional information about why the peer detached.
+
+	// create connection
+	conn, err := amqp.Dial("amqps://my-namespace.servicebus.windows.net", &amqp.ConnOptions{
+		SASLType: amqp.SASLTypePlain("access-key-name", "access-key"),
+	})
+	if err != nil {
+		log.Fatal("Dialing AMQP server:", err)
+	}
+	defer conn.Close()
+
+	ctx := context.TODO()
+
+	// open a session
+	session, err := conn.NewSession(ctx, nil)
+	if err != nil {
+		log.Fatal("Creating AMQP session:", err)
+	}
+
+	// create a sender
+	sender, err := session.NewSender(ctx, "/queue-name", nil)
+	if err != nil {
+		log.Fatal("Creating sender link:", err)
+	}
+
+	// send message
+	err = sender.Send(ctx, amqp.NewMessage([]byte("Hello!")))
+	if err != nil {
+		log.Fatal("Creating AMQP session:", err)
+	}
+
+	// now close the sender
+	err = sender.Close(ctx)
+	if err != nil {
+		log.Fatal("Creating AMQP session:", err)
```

Comment:
```suggestion
		log.Fatal("Creating AMQP sender:", err)
```

Reply:
Other samples don't check the error returned from `Closed()` so I removed this to follow suit.